operator/v1/types_network.go

```diff
@@ -788,3 +815,24 @@ const (
 	// between pods on the cluster network.
 	IPsecModeFull IPsecMode = "Full"
 )
+
+// AdvancedRoutingProvider is a component providing advanced routing capabilities.
+// +kubebuilder:validation:Enum=FRR
+type AdvancedRoutingProvider string
+
+const (
+	// AdvancedRoutingProviderFRR determines FRR is providing advanced routing
+	// capabilities.
+	AdvancedRoutingProviderFRR AdvancedRoutingProvider = "FRR"
+)
+
+// AdvancedRouting describes components and relevant configuration providing
+// advanced routing capabilities.
+type AdvancedRouting struct {
+	// providers of advanced routing capabilities.
+	// +listType=set
```

Comment:
Do not use listType Set, it has some major issues (like you can't delete from it), so instead use atomic and use CEL to assert the set-ness of the list `self.all(x, self.exists_one(y, x == y)`

Reply:
It complained that the validation cost is too high because the list has no upper bound. Since we only expect one entry for now, I settled with MaxItems.

Reply:
That works and is good, but I'd prefer if we could add this validation as well, so that we don't forget to add it in the future when we expand this list